app/src/main/java/com/example/squareonetestproject/MovieRepository.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.squareonetestproject
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+
+class GenreMapFromNetwork(
+    val genres: List<GenreId2Name>
+) {
+    class GenreId2Name(
+        val id: Int,
+        val name: String
+    )
+}
+
+class UpcomingMoviesResFromNetwork(
+    val page: Int,
+    val results: List<MovieEntity>,
+    val total_pages: Int,
+    val total_results: Int
+) {
+    class MovieEntity(
+        val title: String,
+        val poster_path: String,
+        val genre_ids: List<Int>,
+        val overview: String,
+        val release_date: String,
+        val backdrop_path: String
+    )
+}
+
+class MovieRepository @Inject constructor() {
+    private val tmdb = TMDBapi.api
+
+//    val tmdb: TMDBapiInterface = DaggerMovieRepoComponent.create()
+//    DaggerMovieRepoComponent
+    private suspend fun List<UpcomingMoviesResFromNetwork.MovieEntity>.entityToDetail(): List<MovieDetail> = map {
+        MovieDetail(
+            name = it.title,
+            poster = MovieDetail.ImageRequestFactory(
+                url = "https://image.tmdb.org/t/p/original" + it.poster_path,
+            ),
+            backdrop = MovieDetail.ImageRequestFactory(
+                url = "https://image.tmdb.org/t/p/original" + it.backdrop_path,
+            ),
+            genresList = it.genre_ids.let { ids ->
+                // Get genre map from network
+                try {
+                    CoroutineScope(Dispatchers.IO).async {
+                        tmdb.getGenreMap()
```

Comment:
change it to 
`tmdb.getGenreMap().also{
Log.d("movieRepository", "request genre map")
}` and check `Logcat` tab on Android Studio

Reply:
Take a look on `HttpLoggingInterceptor`. It can be useful too

Reply:
Using HttpLoggingInterceptor, it's pretty nice and neat method to log http request!<3